target/scripts/startup/setup.d/security/rspamd.sh

```diff
@@ -142,14 +149,74 @@ function __rspamd__disable_default_modules
 
   for MODULE in "${DISABLE_MODULES[@]}"
   do
-    __rspamd__enable_disable_module "${MODULE}" 'false'
+    __rspamd__helper__enable_disable_module "${MODULE}" 'false'
   done
 }
 
+# This function sets up intelligent learning of Junk, by
+#
+# 1. enabling auto-learn for the classifier-bayes module
+# 2. setting up sieve scripts that detect when a user is moving e-mail
+#    from or to the "Junk" folder, and learning them as ham or spam.
+function __rspamd__setup_learning
+{
+  if [[ ${RSPAMD_LEARN} -eq 1 ]]
+  then
+    __rspamd__log 'debug' 'Setting up intelligent learning of spam and ham'
+
+    local SIEVE_PIPE_BIN_DIR='/usr/lib/dovecot/sieve-pipe'
+    ln -s "$(type -f -P rspamc)" "${SIEVE_PIPE_BIN_DIR}/rspamc"
+
+    sedfile -i -E 's|(mail_plugins =.*)|\1 imap_sieve|' /etc/dovecot/conf.d/20-imap.conf
+    sedfile -i -E '/^}/d' /etc/dovecot/conf.d/90-sieve.conf
+    cat >>/etc/dovecot/conf.d/90-sieve.conf << EOF
+
+  # From elsewhere to Junk folder
+  imapsieve_mailbox1_name = Junk
+  imapsieve_mailbox1_causes = COPY
+  imapsieve_mailbox1_before = file:${SIEVE_PIPE_BIN_DIR}/learn-spam.sieve
+
+  # From Junk folder to elsewhere
+  imapsieve_mailbox2_name = *
+  imapsieve_mailbox2_from = Junk
+  imapsieve_mailbox2_causes = COPY
+  imapsieve_mailbox2_before = file:${SIEVE_PIPE_BIN_DIR}/learn-ham.sieve
+}
+EOF
+
+    cat >"${SIEVE_PIPE_BIN_DIR}/learn-spam.sieve" << EOF
+require ["vnd.dovecot.pipe", "copy", "imapsieve"];
+pipe :copy "rspamc" ["-h", "127.0.0.1:11334", "learn_spam"];
+EOF
+
+    cat >"${SIEVE_PIPE_BIN_DIR}/learn-ham.sieve" << EOF
+require ["vnd.dovecot.pipe", "copy", "imapsieve"];
+pipe :copy "rspamc" ["-h", "127.0.0.1:11334", "learn_ham"];
+EOF
+
+    sievec "${SIEVE_PIPE_BIN_DIR}/learn-spam.sieve"
+    sievec "${SIEVE_PIPE_BIN_DIR}/learn-ham.sieve"
+  else
+    __rspamd__log 'debug' 'Intelligent learning of spam and ham is disabled'
+  fi
+}
+
+# Sets up greylisting based on the environment variable RSPAMD_GREYLISTING.
+function __rspamd__setup_greylisting
+{
+  if [[ ${RSPAMD_GREYLISTING} -eq 1 ]]
+  then
+    __rspamd__log 'debug' 'Enabling greylisting'
+    sed -i -E "s|(enabled =).*|\1 true;|g" /etc/rspamd/local.d/greylist.conf
```

Comment:
I guess we could use `sedfile` here, but not 100% sure 🤔

Reply:
Done in [3c6156e](https://github.com/docker-mailserver/docker-mailserver/pull/3206/commits/3c6156eb131a58d90d847938f7aebecea401c844)